### DISCUSSION TITLE
Postgress AWS instance.

### DISCUSSION THREAD

**Discussion by n1bor** [ACINQ/eclair#2169] [Node operators]
Was considering moving a node with ~600 channels from SQLITE to Postgress on AWS using RDS.

Anyone got any idea what size instance I will need? Was assuming the smallest possible one should be OK?
db.t4g.micro - so 2CPUs and 1Gig of RAM

**Comment by pm47** [comment]
On AWS, never use anything less than large instances in production (personal rule).

If you are constrained on resources, sqlite is a perfectly valid choices and will work much better than a postgres on a micro instance. We have been using sqlite in production with order of magnitudes more channels.

**Comment by n1bor** [comment]
Was looking at this again. For the fail-over/resilience reasons rather than performance.

I see AWS says we should expect a AZ failover to happen in about 30secs to 1min.
And also a automatic maintenance/upgrade happens in about 1-10mins.

If Eclair is running when either of these happens and it tries to read/write to the DB will it just Exit and need restarting once DB is back online?

**Reply by n1bor** [reply]
OK - tested it myself and it does exit as expected.
My next test I increased the connection timeout to 120 secs so is longer than the RDS failover time.
I then did a failover on RDS (reboot - tick the failover box).
As expected I got some connection pool errors. But as soon as RDS instance re-appeared all was fine.

Would this be advisable in production? Advantage is that Eclair does not crash - but instead recovers. Just not sure would be in a good state?

**Comment by n1bor** [comment]
Am now running an instance with 700 channels on a AWS Postgres RDS db.t3.micro with multi AZ mirroring enabled.

For a few days was accruing CPU credits. But now seems to be just over the allowance so will be charged a bit extra.
Changed a few parameters:
Checkpoint_timeout to 1800
autovacuum_analyze_scale_factor to 0.1
autovacuum_analyze_threshold to 100
autovacuum_vacuume_scale_factor to 0.1
autovacuum_vacuum_threshold to 100
Was to try to reduce cpu below the threashold - but did not help.

Most of CPU is AWS RDS processes!
